### DISCUSSION TITLE
validation.py - KeyError

### DISCUSSION THREAD

**Discussion by inui-sdt2** [MatthiasValvekens/pyHanko#95] [Q&A] [answered]
Dear Matthias-san and all

I get the following error in rare cases when using validation.py. 
What kind of case is the possible causes?
It works if I retry, so I've implemented retry logic now.

/pyhanko/sign/validation.py", line 2540, in _cms_objects_to_streams
web: yield seen[obj_bytes]
web: KeyError: b'0\x82\x03\xed0\x82\x01\xd5\x02\x01\x010\r\x06\t*\x86H\x86\xf7\r\x01\x01\x0c\x05\x000W1\x0b0\t\x06\x03U\x04\x06\x13\x02BE1\x190\x17\x06\x03U\x04\n\x13\x10GlobalSign nv-sa1-0+\x06\x03U\x04\x03\x13$GlobalSign CA for AATL - SHA384 - G4\x17\r220329100021Z\x17\r220405100021Z0\x82\x01\x140,\x02\r\x01\xe9\xd8\xa4\xbb\xb6:>\xbaz\xa8\xf5\x97\x17\r200708163400Z0\x0c0\n\x06\x03U\x1d\x15\x04\x03\n\x01\x050,\x02\r\x01\xec\x1c\x92\x15\x8c^\x1e\xe6\x9b\x82g\x1a\x17\r200708163020
ーーsecond half deleted

**Comment by MatthiasValvekens** [comment, marked as answer]
Hi Inui-san,


Thanks for checking in.

That `KeyError` is actually handled within the normal scope of operations of `_cms_objects_to_streams`, so I'm assuming that the real error is triggered somewhere in [this piece of code](https://github.com/MatthiasValvekens/pyHanko/blob/45204434f4c47eddd2476e9781346f98f3fa09de/pyhanko/sign/validation/dss.py#L137-L143). It might have ended up in a part of the stack trace that was cut off from your post. Could you look into that?

**Reply by inui-sdt2** [reply]
thank you for this advice.I think It's CA side issue. but I will look into details.
